io.openems.edge.evse.chargepoint.alpitronic/src/io/openems/edge/evcs/alpitronic/EvcsAlpitronicImpl.java

```diff
@@ -271,24 +398,521 @@ protected ModbusProtocol defineModbusProtocol() {
+						m(EvcsAlpitronic.ChannelId.SETPOINT_REACTIVE_POWER,
+								new SignedDoublewordElement(this.offset.apply(2)))),
+
+				// Read connector-specific input registers (v1.8 layout)
+				new FC4ReadInputRegistersTask(this.offset.apply(0), Priority.LOW,
+						m(EvcsAlpitronic.ChannelId.RAW_STATUS, new UnsignedWordElement(this.offset.apply(0))),
+						m(EvcsAlpitronic.ChannelId.CHARGING_VOLTAGE,
+								new UnsignedDoublewordElement(this.offset.apply(1)), SCALE_FACTOR_MINUS_2),
+						m(EvcsAlpitronic.ChannelId.CHARGING_CURRENT, new UnsignedWordElement(this.offset.apply(3)),
+								SCALE_FACTOR_MINUS_2),
+						m(EvcsAlpitronic.ChannelId.RAW_CHARGE_POWER,
+								new UnsignedDoublewordElement(this.offset.apply(4))),
+						m(EvcsAlpitronic.ChannelId.CHARGED_TIME, new UnsignedWordElement(this.offset.apply(6))),
+						m(EvcsAlpitronic.ChannelId.CHARGED_ENERGY, new UnsignedWordElement(this.offset.apply(7)),
+								SCALE_FACTOR_MINUS_2).onUpdateCallback(e -> {
+									if (e == null) {
+										return;
+									}
+									if (e == 0) {
+										switch (this.getStatus()) {
+										case UNDEFINED:
+										case NOT_READY_FOR_CHARGING:
+										case STARTING:
+											this._setEnergySession(0);
+											return;
+										case CHARGING:
+										case CHARGING_REJECTED:
+										case ENERGY_LIMIT_REACHED:
+										case ERROR:
+										case READY_FOR_CHARGING:
+											// Ignore 0 value
+											return;
+										}
+									}
+									this._setEnergySession(e * 10);
+								}),
+						m(EvcsAlpitronic.ChannelId.EV_SOC, new UnsignedWordElement(this.offset.apply(8)),
+								SCALE_FACTOR_MINUS_2),
+						m(EvcsAlpitronic.ChannelId.CONNECTOR_TYPE, new UnsignedWordElement(this.offset.apply(9))),
+						m(EvcsAlpitronic.ChannelId.EV_MAX_CHARGING_POWER,
+								new UnsignedDoublewordElement(this.offset.apply(10))),
+						m(EvcsAlpitronic.ChannelId.EV_MIN_CHARGING_POWER,
+								new UnsignedDoublewordElement(this.offset.apply(12))),
+						m(EvcsAlpitronic.ChannelId.VAR_REACTIVE_MAX,
+								new UnsignedDoublewordElement(this.offset.apply(14))),
+						m(EvcsAlpitronic.ChannelId.VAR_REACTIVE_MIN,
+								new UnsignedDoublewordElement(this.offset.apply(16)), INVERT),
+						// Gap at offset 17
+						new DummyRegisterElement(this.offset.apply(17)),
+						// VID (Vehicle ID) at offset 118-121 (4 registers = 8 bytes)
+						m(EvcsAlpitronic.ChannelId.VID, new StringWordElement(this.offset.apply(18), 4)),
+						// idTag (OCPP ID) at offset 122-131 (10 registers = 20 chars)
+						m(EvcsAlpitronic.ChannelId.ID_TAG, new StringWordElement(this.offset.apply(22), 10))));
+
+		this.addCalculatePowerListeners();
+		this.addStatusListener();
+		this.addVersionListener();
+
+		return modbusProtocol;
+	}
+
+	/**
+	 * Defines modbus protocol for firmware version 2.3.x.
+	 *
+	 * @return the ModbusProtocol configured for v2.3.x register mapping
+	 */
+	private ModbusProtocol defineModbusProtocolV23() {
```

Comment:
Many tasks are currently redundant/duplicated. If most Tasks are "copied" and differences are only in certain tasks/blocks, we could add registers like we do in `GoodWe`